java-client/src/integration-test/java/com/cumulocity/sdk/client/notification/RealtimeMultiSubNotificationIT.java

```diff
@@ -0,0 +1,194 @@
+
+    InventoryApi inventoryApi = platform.getInventoryApi();
+    DeviceControlApi operationsApi = platform.getDeviceControlApi();
+
+    @Test
+    public void shouldNotReceiveDuplicatedOperations() throws InterruptedException, ExecutionException {
+        ExecutorService executor = Executors.newFixedThreadPool(CONCURRENT_JOBS);
+
+        List<Callable<Boolean>> jobs = new ArrayList<>();
+        for (int jobNo = 0; jobNo < CONCURRENT_JOBS; ++jobNo) {
+            jobs.add(new SubscriberJob(jobNo));
+        }
+
+        boolean failed = false;
+        for (Future<Boolean> result : executor.invokeAll(jobs)) {
+            if (!result.get()) {
+                failed = true;
+                break;
+            }
+        }
+
+        assertThat(failed)
+                .withFailMessage("Test failed. See the logs to investigate the issue.")
+                .isFalse();
+    }
+
+    private class SubscriberJob implements Callable<Boolean> {
+        private final int jobNo;
+        private final Subscriber<String, RealtimeMessage> subscriber;
+
+        public SubscriberJob(int jobNo) {
+            this.jobNo = jobNo;
+            this.subscriber = SubscriberBuilder.<String, RealtimeMessage>anSubscriber()
+                    .withParameters(platform)
+                    .withEndpoint(REALTIME)
+                    .withSubscriptionNameResolver(name -> name)
+                    .withDataType(RealtimeMessage.class)
+                    .build();
+        }
+
+        @Override
+        public Boolean call() throws InterruptedException {
+            ManagedObjectRepresentation device = inventoryApi.create(aMoDevice());
+
+            AtomicInteger opCounter = new AtomicInteger(0);
+            List<Subscription<String>> subscriptions = subscribe(device.getId(), opCounter);
+
+            operationsApi.create(aDeviceOperation(device.getId()));
+
+            try {
+                // check counter as fast as possible and then wait 1s to verify it didn't go up
+                await().atMost(TEN_SECONDS).until(() -> opCounter.get() == 1);
+                await().pollDelay(ONE_SECOND).atMost(TEN_SECONDS).until(() -> opCounter.get() == 1);
+            } catch (ConditionTimeoutException ex) {
+                log.error("Subscription test failed with notifications count {} (should be 1)", opCounter.get());
+                return false;
+            }
+
+            subscriptions.forEach(Subscription::unsubscribe);
+            subscriber.disconnect();
+
+            return opCounter.get() == 1;
+        }
+
+        private List<Subscription<String>> subscribe(GId deviceId, AtomicInteger opCounter) throws InterruptedException {
+            CountDownLatch subOpLatch = new CountDownLatch(5);
+
+            List<Subscription<String>> subscriptions = Stream.of("measurements", "alarms", "events", "managedobjects")
+                    .map(channel -> subscriber.subscribe(
+                            "/" + channel + "/*",
```

Comment:
Why are you subscribing to `*` and then filter by response payload instead of just subscribing by specific device id?

Reply:
That was the scenario from Apama team so I sticked to it.